File: problem_002/002.js
// Even Fibonacci numbers
// find sum of even-valued terms whose values do not exceed four million

//gets nth fibonacci value
var fibonacci = function(n) {
    return (n === 1 || n === 2) ? n : fibonacci(n - 1) + fibonacci(n - 2);
};

//get sum of even fibonacci values from 1 to max value
var sum_fib_evens = function(max) {
  var total = 0;
  var n = 1;
  while (fibonacci(n) < max) {
    total += (fibonacci(n) < max && fibonacci(n) % 2 === 0) ? fibonacci(n) : 0;
    n++;
  }
  return total;
};

console.log(sum_fib_evens(4000000));

File: problem_002/002.rb
#Even Fibonacci numbers
#find sum of even-valued terms whose values do not exceed four million

#get nth fibonacci number
def fibonacci(n)
  if n <= 1
    return n
  else
    return fibonacci(n - 1) + fibonacci(n - 2)
  end
end

#get sum of even fibonacci values from 1 to max
def sum_fib_evens(max)
  sum = 0
  n = 1
  while fibonacci(n) < max
    if fibonacci(n).even?
      sum += fibonacci(n)
    end
    n += 1
  end
  return sum
end


